src/core/CorePool.ts

```diff
@@ -100,10 +101,29 @@ export class CorePool {
     tickUpper: number,
     amount: JSBI
   ): { amount0: JSBI; amount1: JSBI } {
-    // TODO
+    if (JSBI.lessThanOrEqual(amount, ZERO)) {
+      console.error("[Error]: mint amount should greater than 0");
+    }
+
+    let amount0 = ZERO;
+    let amount1 = ZERO;
+
+    if (JSBI.greaterThan(amount, ZERO)) {
+      let positionStep = this.modifyPosition(
+        recipient,
+        tickLower,
+        tickUpper,
+        amount
+      );
+
+      amount0 = positionStep.amount0;
+      amount1 = positionStep.amount1;
+    } else {
+      console.error("[Error]: amount need greater than 0");
```

Comment:
The same as above. If mint condition is not satisfied, then user as caller should handle that.